datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/KafkaPositionTracker.java

```diff
@@ -541,13 +573,63 @@ protected void runOneIteration() {
         }
       }));
 
-      // Query the broker for its offsets for those partitions
       try {
-        _kafkaPositionTracker.queryBrokerForLatestOffsets(_consumer, partitionsNeedingUpdate);
+        updatePartitionLeadershipMap();
```

Comment:
Since this is a performance optimization potentially have negative performance impact , could you have a config to enable/disable this?